### PR TITLE
[hail] release 0.2.21

### PR DESCRIPTION
**Pull request by danking**

### CODE REVIEW

File: hail/python/hail/docs/change_log.md [hunk 1]
```diff
@@ -1,5 +1,35 @@
...
+### Documentation
+- (hail#6904) Fix documentation of `sparse_split_multi`
+- (hail#6789) Fix visual bugs on website and docs
+- (hail#6914) Fix the "edit this page" link
+
+### `hailctl dataproc`
+- (hail#6904) Add `--dry-run` option to `submit`.
+- (hail#6951) Fix `--max-idle` and `--max-age` arguments to `start`
+- (hail#6919) Add `--update-hail-version` to `modify`
+
```
Comment (tpoterba) [inline hail/python/hail/docs/change_log.md]: add a `----` here

**Comment by tpoterba** [comment]
can we merge this after #6969 goes in? That resolves a pretty big performance regression in stats/mean/corr/hist aggregators.

File: hail/python/hail/docs/change_log.md [hunk 2]
```diff
@@ -1,5 +1,35 @@
 # Change Log
 
+## 0.2.21
+
+Release 2019-08-30
+
+### Correctness
+- (hail#6945) Fix `expand_types` to preserve the order of the data in a table
```
Comment (tpoterba) [inline hail/python/hail/docs/change_log.md]: to clarify: "to preserve ordering by key". Also note `to_pandas` and `to_spark` are affected.

#6958 is not a correctness fix.

File: hail/python/hail/docs/change_log.md [hunk 3]
```diff
@@ -1,5 +1,35 @@
 # Change Log
 
+## 0.2.21
+
+Release 2019-08-30
```
Comment (tpoterba) [inline hail/python/hail/docs/change_log.md]: typo: released

File: hail/python/hail/docs/change_log.md [hunk 4]
```diff
@@ -1,5 +1,35 @@
...
+
+### Correctness
+- (hail#6945) Fix `expand_types` to preserve the order of the data in a table
+- (hail#6958) Fix stack overflow errors when counting the result of a `Table.union`
+
+### Functionality
+- (hail#6913) Teach `sparse_split_multi` how to filter changed loci
```
Comment (tpoterba) [inline hail/python/hail/docs/change_log.md]: sparse_split_multi is experimental and not really publicly exposed, I'd remove this.

File: hail/python/hail/docs/change_log.md [hunk 5]
```diff
@@ -1,5 +1,35 @@
...
+- (hail#6903) Teach `hl.range` to treat a single argument as `0..N`
+- (hail#6903) Teach `BlockMatrix` how to `checkpoint`
+
+### Performance
+- (hail#6896) Improve performance of `sparse_split_multi`
+- (hail#6895) Improve performance of `hl.import_bgen(...).count()`
+- (hail#6948) Fix performance bug in `BlockMatrix` filtering functions
+- (hail#6943) Improve scaling of `Table.union`
+
+### Documentation
```
Comment (tpoterba) [inline hail/python/hail/docs/change_log.md]: I haven't been including docs changes, to make changelog construction easier.

File: hail/python/hail/docs/change_log.md [hunk 6]
```diff
@@ -1,5 +1,35 @@
...
 
+## 0.2.21
+
+Release 2019-08-30
+
+### Correctness
+- (hail#6945) Fix `expand_types` to preserve the order of the data in a table
+- (hail#6958) Fix stack overflow errors when counting the result of a `Table.union`
+
+### Functionality
```
Comment (tpoterba) [inline hail/python/hail/docs/change_log.md]: we've generally been calling this "New features"

File: hail/python/hail/docs/change_log.md [hunk 7]
```diff
@@ -1,5 +1,35 @@
 # Change Log
 
+## 0.2.21
+
+Release 2019-08-30
+
+### Correctness
```
Comment (tpoterba) [inline hail/python/hail/docs/change_log.md]: We've been calling this section "Bug fixes"

**Comment by danking** [comment]
I'm good with waiting, ping here when ready

File: hail/python/hail/docs/change_log.md (hunk 2 above)
Comment (danking) [reply hail/python/hail/docs/change_log.md]: Error is a kind of incorrect 😉

**Comment by tpoterba** [comment]
Arcturus has been without internet and unable to review #6969 - OK if we release tomorrow?

**Comment by danking** [comment]
I'd be delighted to do so.

**Comment by danking** [comment]
I'll rebase and add any missing notes.

**Comment by tpoterba** [comment]
actually, let's just release now. We can make another release at the end of the week -- there are a couple of major performance improvements queued up.

**Comment by danking** [comment]
OK

**Comment by tpoterba** [comment]
oops I did tell Beryl we'd get #6978 in the release, though

**Comment by akotlar** [comment]
and a new website design!

**Comment by danking** [comment]
This branch is up to date with master's `69ba846a3`. Let me know when #6978 lands and I'll update again. Maybe nudge @iitalics to review that?

File: hail/python/hail/docs/change_log.md [hunk 8]
```diff
@@ -1,5 +1,32 @@
 # Change Log
 
+## 0.2.21
+
+Released 2019-08-30
```
Comment (tpoterba) [inline hail/python/hail/docs/change_log.md]: change date to today?

File: hail/python/hail/docs/change_log.md [hunk 9]
```diff
@@ -1,5 +1,32 @@
...
+### New features
+- (hail#6856) Teach `hl.agg.counter` to weigh each value differently.
+- (hail#6903) Teach `hl.range` to treat a single argument as `0..N`.
+- (hail#6903) Teach `BlockMatrix` how to `checkpoint`.
+
+### Performance
+- (hail#6895) Improve performance of `hl.import_bgen(...).count()`.
+- (hail#6948) Fix performance bug in `BlockMatrix` filtering functions.
+- (hail#6943) Improve scaling of `Table.union`.
+- (hail#6980) Improve performance of `split_multi_hts`
```
Comment (tpoterba) [inline hail/python/hail/docs/change_log.md]: period. Also mention that it's about a 40% improvement.

**Pull request merged**